### ISSUE TITLE
Slow Tab perfomance on Webview

### ISSUE THREAD

**Issue by riccardo-lomazzi** [material-components/material-components-web-react#453]
I'm trying to implement these React components in my webapp, that renders on Android with a Webview. 
I'm not having issues on a mid-budget Android Phone, but on an older phone like the Huawaei G Play Mini, the tabs are very resource-heavy, and make the whole experience sluggish so much it takes even 2 seconds before registering an input. I've even tried to update the Webview package, with no luck. 
Is there anything I should consider before scrapping these components?

**Comment by moog16** [comment]
@riccardo-lomazzi could you please share a codesandbox of your working code? It would help to see the code you're working with.

In the meantime I will try to reproduce on my end.

**Comment by moog16** [comment]
@riccardo-lomazzi I tested to see if extra renders were trigger from a tab click. This triggered 2 render cycles. One change from `props.activeIndex` changing, and one from the anonymous function `handleActiveIndexUpdate`. I made `handleActiveIndexUpdate` a component method, which reduced this one render cycle. Documented in [this PR](https://github.com/material-components/material-components-web-react/pull/465/files#diff-d798af0b348f0afb61b6e03068d92cd9R41).

The only other thing I can think of is that MDC Web's foundation is sluggish. During click it does have a call to `computeIndicatorClientRect()`. Could this be causing your issue?
